sdk/confidentialledger/azure-confidentialledger/azure/confidentialledger/_client.py

```diff
@@ -0,0 +1,480 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+import time
+
+from azure.core.tracing.decorator import distributed_trace
+
+from ._enums import LedgerUserRole, TransactionState
+from ._generated_ledger.models import ConfidentialLedgerQueryState
+from ._models import (
+    AppendResult,
+    Constitution,
+    Consortium,
+    ConsortiumMember,
+    EnclaveQuote,
+    LedgerEnclaves,
+    LedgerEntry,
+    LedgerUser,
+    TransactionReceipt,
+    TransactionStatus,
+)
+from ._shared import (
+    ConfidentialLedgerCertificateCredential,
+    ConfidentialLedgerClientBase,
+)
+
+try:
+    from typing import TYPE_CHECKING
+except ImportError:
+    TYPE_CHECKING = False
+
+if TYPE_CHECKING:
+    from azure.core.credentials import TokenCredential
+    from azure.core.paging import ItemPaged
+    from typing import Any, Union
+
+
+class ConfidentialLedgerClient(ConfidentialLedgerClientBase):
+    """A client for putting data into and querying data from the Confidential Ledger service.
+
+    The `transport` parameter is typically accepted by Azure SDK clients to provide a custom
+    transport stage in the pipeline. Since this client makes modifications to the default transport,
+    using a custom transport will override and remove the following functionality:
+        1) Authentication using a client certificate.
+        2) TLS verification using the Confidential Ledger TLS certificate.
+
+    :param str ledger_url: URL of the Confidential Ledger service.
```

Comment:
According to track 2 guidelines, I think this parameter should be called `endpoint` (cc @lmazuel who I just learned this from 😄 )